### PR TITLE
add TrkDtc fragment and the corresponding type

### PR DESCRIPTION
**Pull request by pavel1murat**
also comment out a few HIDE_FROM_ROOT clauses: 

- after the TRACE includes got protected, this is no longer necessary. 
- ROOT dictionaries are not denerated anyway

### CODE REVIEW

File: artdaq-core-mu2e/Data/ArtFragment.hh [hunk 1]
```diff
@@ -4,9 +4,9 @@
 #include "artdaq-core/Data/Fragment.hh"
 #include "artdaq-core/Data/dictionarycontrol.hh"
 #include "cetlib_except/exception.h"
-#if HIDE_FROM_ROOT
+// #if HIDE_FROM_ROOT
```
Comment (eflumerf) [inline artdaq-core-mu2e/Data/ArtFragment.hh]: I would prefer removing unneeded code completely, rather than simply commenting.

Comment (pavel1murat) [reply artdaq-core-mu2e/Data/ArtFragment.hh]: Makes sense. I commented it out  only to have an easy step back, in case that was necessary.

Comment (pavel1murat) [reply artdaq-core-mu2e/Data/ArtFragment.hh]: comments removed

**Review by eflumerf — APPROVED** [review]
LGTM

**Pull request merged**